sdk/core/Azure.Core/src/Pageable.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading;
+
+namespace Azure
+{
+    public abstract class Pageable<T> : IEnumerable<T> where T : notnull
```

Comment:
We need to enable a scenario where item enumerator (not paging) starts from a continuation token. Is this supported?

Reply:
I think we said that if continuation token or size hint gets involved you have to use paging apis because there is no way to get continuation token out of iterator if you are not doing paging.

Reply:
Cosmos team requested this. They API does not need to be public, but if somebody inherits from the abstract collection, they should be able to start it from a continuation.

Reply:
I think it's possible.